myapp/views.py

```diff
@@ -61,3 +65,97 @@ def homework4(request):
     content['task6'] = Comment.objects.filter(content_type_id=article_model_id).order_by('date', '-article__author')[:2]
 
     return render(request, 'homework4.html', content)
+
+
+def check_requirements_form_view(request):
+    if request.method == 'POST':
+        form = CheckRequirementsForm(request.POST)
+        if form.is_valid():
+            name = request.POST['name']
+            sex = request.POST['sex']
+            age = int(request.POST['age'])
+            level = request.POST['english_level']
+
+            is_fit = (sex == 'm' and age >= 20 and level in ['C1', 'C2']) or \
+                     (sex == 'f' and age >= 22 and level in ['B2', 'C1', 'C2'])
+            content = {'name': name, 'fit': is_fit, 'title': "Congratulations!" if is_fit else "Don't worry!"}
+
+            return render(request, 'form_answer.html', content)
+    else:
+        initial = {'name': (request.user.first_name or request.user.username) if request.user.is_authenticated else ''}
+        form = CheckRequirementsForm(initial=initial)
+        # we fill the name_fild of form by name/username of logged user
+    content = {'title': 'Check if you fit', 'form': form}
+
+    return render(request, 'requirements_form.html', content)
+
+
+def user_login(request):
+    if request.method == 'POST':
+        form = AuthForm(request.POST)
+        if form.is_valid():
+            login(request, form.user)
+            return HttpResponseRedirect('/')
+    else:
+        form = AuthForm()
+
+    return render(request, 'login.html', {'title': 'Authentication', 'form': form})
+
+
+def user_logout(request):
+    logout(request)
+    return HttpResponseRedirect('/login')
+
+
+def user_register(request):
+    if request.method == 'POST':
+        form = RegisterForm(request.POST)
+        if form.is_valid():
+            user = User.objects.create_user(username=request.POST['username'],
```

Comment:
Please get data from form.cleaned_data If you will change form and form will be valid but data will be in request you can create an incorrect user or get an exception

Reply:
totally agree, fixed